sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/ShareAsyncApiTests.java

```diff
@@ -376,18 +387,22 @@ public void setPremiumProperties() {
             = Arrays.asList(ShareRootSquash.ALL_SQUASH, ShareRootSquash.NO_ROOT_SQUASH, ShareRootSquash.ROOT_SQUASH);
 
         for (ShareRootSquash rootSquash : rootSquashes) {
-            ShareAsyncClient premiumShareClient = Objects.requireNonNull(premiumFileServiceAsyncClient
-                .createShareWithResponse(generateShareName(),
+            String shareName = generateShareName();
+            Mono<ShareAsyncClient> premiumShareClientMono = premiumFileServiceAsyncClient
+                .createShareWithResponse(shareName,
                     new ShareCreateOptions().setProtocols(new ShareProtocols().setNfsEnabled(true)), null)
-                .block()).getValue();
-            premiumShareClient.setProperties(new ShareSetPropertiesOptions().setRootSquash(rootSquash)).block();
-            StepVerifier.create(premiumShareClient.getProperties())
+                .map(Response::getValue);
+
+            StepVerifier
+                .create(premiumShareClientMono.flatMap(premiumShareClient -> premiumShareClient
+                    .setProperties(new ShareSetPropertiesOptions().setRootSquash(rootSquash))
+                    .then(premiumShareClient.getProperties())))
                 .assertNext(it -> assertEquals(rootSquash, it.getRootSquash()))
                 .verifyComplete();
-
-            //cleanup
-            premiumShareClient.delete().block();
         }
+
+        //cleanup
+        premiumFileServiceAsyncClient.getShareAsyncClient(shareName).delete().block();
```

Comment:
we want to make sure we add this line inside of the for loop to delete each share after its created